src/hooks/useDebounceSearch.ts

```diff
@@ -15,9 +15,13 @@ export const useDebounceSearch = () => {
 
   const debouncedSearch = useMemo(
     () =>
-      debounce((value: string) => {
-        setSearchQuery(value.toLowerCase());
-      }, SEARCH_DEBOUNCE_TIME),
+      debounce(
+        (value: string) => {
+          setSearchQuery(value.toLowerCase());
+        },
+        SEARCH_DEBOUNCE_TIME,
+        { maxWait: SEARCH_DEBOUNCE_TIME * 3 }
```

Comment:
For some reason, the execution of the denounced func was delayed for unreasonable amount of time (>5 seconds)

Reply:
Was this while experiencing other performance issues/where the browser was already struggling? If so, might have been related to that more than our use of `debounce` here?

Reply:
If we keep this, might be worth extracting `SEARCH_DEBOUNCE_TIME * 3` as a `const` in settings as well.